### ISSUE TITLE
Default delete button behavior

### ISSUE THREAD

**Issue by Destroy666x** [mybb/mybb#1894]
As per comment here: https://github.com/mybb/mybb/issues/1882 I think there should be a setting for default delete button behavior for admins/mods. Right now in 1.8.4 it's forced to soft delete without core edits, which 2-3 people complained about. Should be fairly easy to add.

**Comment by Cu8eR** [comment]
+1, I agree

**Comment by Tbo29** [comment]
Also agree

**Comment by JN-Jones** [comment]
While I agree with the idea I'm not sure whether we want to really add more features @mybb/developers @mybb/management

**Comment by justinsoltesz** [comment]
From a user experience point of view I don't think this should be changed. If I'm not mistaken the only way to permanently delete a post currently when soft delete is enabled is to use the mod tools -- changing that (even if it's just a setting) would lead to inconsistencies between the expected result for normal users vs. mods/admins, as well as for users who only moderate certain forums. Also, if you want to permanently delete a post, you can just use the mod tools instead of the delete button.

**Comment by Stefan-ST** [comment]
Providing both options (like we do for editing posts) would solve this.

**Comment by justinsoltesz** [comment]
Agreed, that would be a better solution.

**Comment by euantorano** [comment]
I agree @Stefan-ST's suggestion would work best, if somebody's willing to write it. I'd rather not add new features to 1.8, but this should be a minor change.

**Comment by effone** [comment]
@euantorano are we still willing to add this in 1.8?
This is not a minor change as it seems, but I can do it if required ...

**Comment by euantorano** [comment]
If it turns out to be a big job, I’m willing to delay until 1.9.

**Comment by effone** [comment]
Is this still in our to do list?
I have a better idea to accomplish this without / very minimal (just inserting language / settings variables) template edit.
@euantorano

**Comment by Ben-MyBB** [comment]
I'd say so @effone if you are willing to work on it :)

**Comment by Sama34** [comment]
If you can achieve this without much edit it could be acceptable for 1.8 I suppose this won't break themes either as the former behavior should still work.

**Comment by Eldenroot** [comment]
> Is this still in our to do list?
> I have a better idea to accomplish this without / very minimal (just inserting language / settings variables) template edit.
> @euantorano

Would be nice, because soft deletion is not always the best solution and you wanna delete permanently.

**Comment by euantorano** [comment]
If someone is willing to do the work and has a plan, I'm happy for them to go ahead. We have to be careful not to break themes or plugins too much, but otherwise we should be fine.

**Comment by Sama34** [comment]
Ideally all moderation actions should be able to be inserted as buttons, but that would be better off to be worked for on 1.9

**Comment by Eldenroot** [comment]
What about to add one more setting for permanently delete threads and permanently delete posts? This settings would be only for mod/admin and will change the behaviour of the button. This is simple change. I can create a PR for this if you agree.

I do not know what @effone wanna add or what is his idea.

Anyway this is quite often mentioned in MyBB community board.

Let me know, if it will not be too complicated (your ideas) I would like to add it myself.

**Comment by Sama34** [comment]
I think that for 1.9 all mod actions should be embeddable in posts, but if toggling the delete button functionality is easy with just one extra setting then I think it is worth it as a feature.

**Comment by Eldenroot** [comment]
One extra setting would be fine... but I need your (MyBB dev team) final decision before I will work on it.

**Comment by euantorano** [comment]
One extra setting is fine, please go ahead.

**Comment by effone** [comment]
I have already worked on it. There is no requirement of additional settings.
What My PR does is if the soft delete feature is disabled it goes for permanent deletion, if enabled it shows a dropdown to choose the action from, exactly as @Stefan-MyBB suggested.

This requires editing delete button template, so I didn't push the PR.

Oh, and on a no-js fallback situation it goes for direct soft delete, like it does now.

**Comment by Eldenroot** [comment]
@effone - I think one template edit is fine, we do template edits in almost every version and this could be a really good and useful addition. My idea was to add one more setting in ACP (similar to edit by admins/mods but for delete button).

EDIT: And I think the AJAX is broken for deleting posts... page must be reloaded manually to see changes... ajax for restoring posts works fine (at least here on my local copy, maybe it is an invalid report)...

**Comment by effone** [comment]
Tell me about the new setting and its exact purpose. I can't see any.
AJAX delete works fine.

**Comment by Eldenroot** [comment]
I tried fresh install of MyBB and AJAX for deleting post is not working here---I have to refresh the page to see the different background color of deleted post etc.

For settings - in ACP - posting - check edit setting for mods/admins. Something like this could be also for soft/perma delete.

**Comment by Eldenroot** [comment]
@effone - anyway you implementation would be better - in a pop up window user could select desired option... let me knwo, I will help with testing.
